sklearn/tests/test_impute.py

```diff
@@ -490,3 +493,602 @@ def test_mice_additive_matrix():
                           random_state=rng).fit(X_train)
     X_test_est = imputer.transform(X_test)
     assert_allclose(X_test_filled, X_test_est, atol=0.01)
+
+
+#############################################################################
+# BEGIN KNNIMPUTER TEST
+
+
+def test_knn_imputation_shape():
+    # Verify the shapes of the imputed matrix for different weights and
+    # number of neighbors.
+    n_rows = 10
+    n_cols = 2
+    X = np.random.rand(n_rows, n_cols)
+    X[0, 0] = np.nan
+
+    for weights in ['uniform', 'distance']:
+        for n_neighbors in range(1, 6):
+            imputer = KNNImputer(n_neighbors=n_neighbors, weights=weights)
+            X_imputed = imputer.fit_transform(X)
+            assert_equal(X_imputed.shape, (n_rows, n_cols))
+
+
+def test_knn_imputation_zero():
```

Comment:
The fact that this only does validation tests should probably be mentioned in the test name